File: binary-search/sorted-list-position.cpp
int Solution::searchInsert(vector<int> &A, int B) {
    int l = 0, r = A.size() - 1, mid;
    while(l <= r){
        mid = (l + r)/2;
        if(A[mid] > B) r = mid - 1;
        else if(A[mid] < B) l = mid + 1;
        else return mid;
    }
    return l;
}

File: binary-search/sorted-list-position.py
class Solution:
    def searchInsert(self, A, B):
        low = 0
        high = len(A) - 1

        while low <= high:
            mid = low + (high - low)//2

            if A[mid] == B: return mid
            elif A[mid] > B: high = mid - 1
            else: low = mid + 1

        return low

